stl/inc/stdatomic.h

```diff
@@ -9,6 +9,10 @@
 #include <yvals.h>
 #if _STL_COMPILER_PREPROCESSOR
 
+#ifndef __cplusplus
+#error <stdatomic.h> is not yet supported when compiling as C, but this is planned for a future release.

```

Comment:
Are we certain we're prepared to make this promise? (Ok to merge as-is, but I'd hate to have to take this back.)

Reply:
Yes - we've already announced this promise in ["C11 and C17 Standard Support Arriving in MSVC"](https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/), published September 2020:

> While there is currently no support for any C11 optional features, we are committed to providing the most impactful optional features in future releases. Atomic and threading support are on our roadmap.

I confirmed this with Daniel Griffing, compiler front-end dev lead, who pointed me to this blog post, verified that this is still on the list of features to implement, and approved this phrasing for the error message (since this is essentially a promise from the FE team).